webforj-plugins/webforj-minify/webforj-minify-foundation/src/main/java/com/webforj/minify/processor/AssetAnnotationProcessor.java

```diff
@@ -0,0 +1,362 @@
+    Map<? extends ExecutableElement, ? extends AnnotationValue> values =
+        processingEnv.getElementUtils().getElementValuesWithDefaults(mirror);
+
+    for (Map.Entry<? extends ExecutableElement, ? extends AnnotationValue> entry : values
+        .entrySet()) {
+      String paramName = entry.getKey().getSimpleName().toString();
+
+      // All webforJ annotations use 'value' parameter only
+      if ("value".equals(paramName)) {
+        processAnnotationValue(entry.getValue().getValue(), type, sourceClass);
+      }
+    }
+  }
+
+  private void processAnnotationValue(Object value, String type, String sourceClass) {
+    if (value instanceof String url) {
+      addResourceIfNotEmpty(url, type, sourceClass);
+    } else if (value instanceof List) {
+      processAnnotationValueList(value, type, sourceClass);
+    }
+  }
+
+  private void processAnnotationValueList(Object value, String type, String sourceClass) {
+    @SuppressWarnings("unchecked")
+    List<? extends AnnotationValue> list = (List<? extends AnnotationValue>) value;
+    for (AnnotationValue av : list) {
+      String url = av.getValue().toString();
+      addResourceIfNotEmpty(url, type, sourceClass);
+    }
+  }
+
+  private void addResourceIfNotEmpty(String url, String type, String sourceClass) {
+    if (!url.isEmpty() && isLocalResource(url, sourceClass)) {
+      resources.add(new ResourceEntry(url, type, sourceClass));
+      processingEnv.getMessager().printMessage(Diagnostic.Kind.NOTE,
+          "Discovered " + type + " asset: " + url + " in " + sourceClass);
+    } else if (!url.isEmpty() && !isLocalResource(url, sourceClass)) {
+      processingEnv.getMessager().printMessage(Diagnostic.Kind.NOTE,
+          "Skipping external resource (cannot be minified): " + url);
+    }
+  }
+
+  private boolean isLocalResource(String url, String sourceClass) {
+    // Only process local resources that can be minified
+    // External URLs (http://, https://) cannot be minified - they're hosted on CDNs
+    // Icons (icons://) are images, not CSS/JS files
+    if (url.startsWith("http://") || url.startsWith("https://")) {
+      return false; // External CDN resources
+    }
+    if (url.startsWith("icons://")) {
+      return false; // Icon resources (images, not minifiable)
+    }
+
+    // Check for valid protocols
+    if (url.contains("://")) {
+      // Has a protocol - must be ws:// or context://
+      if (!url.startsWith("ws://") && !url.startsWith("context://")) {
+        processingEnv.getMessager().printMessage(Diagnostic.Kind.WARNING,
+            "Unknown protocol in asset URL: " + url + " in " + sourceClass
+                + " (only ws://, context:// are supported for minification)");
+        return false;
+      }
+    } else {
+      // No protocol - webforJ passes these through unchanged to the browser
+      // They cannot be reliably mapped to filesystem paths for minification
+      processingEnv.getMessager().printMessage(Diagnostic.Kind.WARNING,
```

Comment:
the `ws://` protocol does not work for @InlineJavaScript and @InlineStyleSheet, only `context://`